src/ml_board/backend/restful_api/data_access.py

```diff
@@ -236,6 +240,70 @@ def get_experiment_config(self, grid_search_id: str, experiment_id: str, config_
         else:
             raise InvalidPathError(f"File path {requested_full_path} is not safe.")
 
+    def get_checkpoint_dict_epoch(self, grid_search_id: str, experiment_id: str, epoch: str):
+        """
+        Fetch all checkpoint resource pickle files from event storage
+        given the epoch, experiment ID & grid search ID.
+
+        :params:
+             grid_search_id (str): Grid Search ID
+             experiment_id (str): Experiment ID
+             epoch (str): Epoch number
+
+        :returns: List of Checkpoint files
+        """
+        requested_full_path = os.path.realpath(
+            os.path.join(self.top_level_logging_path, str(grid_search_id), str(experiment_id), str(epoch))
+        )
+
+        if FileDataAccess.is_safe_path(base_dir=self.top_level_logging_path, requested_path=requested_full_path):
+            response = []
+            checkpoints = []
+            files = FileDataAccess.get_checkpoint_files(requested_full_path, base_path=self.top_level_logging_path)
+            for file_num in range(len(files)):
+                split = os.path.normpath(files[file_num]).split(os.sep)
+                checkpoints.append(os.path.basename(split[3]).split(".")[0])
+                if file_num == len(files) - 1:
```

Comment:
is this if clause really necessary? We could just remove line 266 and put line 276 outside of the loop, no?